src/constructor.io/client/modules/Catalog.cs

```diff
@@ -380,5 +381,504 @@ public async Task<string> DeleteItemGroups(ItemGroupsRequest itemGroupsRequest)
+            }
+
+            throw new ConstructorException("DeleteFacetConfig response data is malformed");
+        }
+
+        // Facet Options
+        internal string CreateFacetOptionsUrl([Optional] string section, string facetGroupName, [Optional] string facetOptionValue, [Optional] Hashtable queryParams)
+        {
+            List<string> paths = new List<string> { "v1", "facets", facetGroupName, "options" };
+
+            if (facetOptionValue != null)
+            {
+                paths.Add(facetOptionValue);
+            }
+
+            if (queryParams == null)
+            {
+                queryParams = new Hashtable();
+            }
+
+            if (!string.IsNullOrEmpty(section))
+            {
+                queryParams.Add(Constants.SECTION, section);
+            }
+
+            Dictionary<string, bool> omittedQueryParams = new Dictionary<string, bool>()
+            {
+                { "_dt", true },
+                { "c", true },
+            };
+
+            string url = MakeUrl(this.Options, paths, queryParams, omittedQueryParams);
+
+            return url;
+        }
+
+        /// <summary>
+        /// Creates a Facet Option given a facet group name.
+        /// <param name="facetOption">New Facet Option to be created.</param>
+        /// <param name="facetGroupName">Facet Group where the Facet Option should be created.</param>
+        /// <param name="section">Section in which the facet is defined.</param>
+        /// </summary>
+        public async Task<FacetOption> CreateFacetOption(FacetOption facetOption, string facetGroupName, [Optional] string section)
+        {
+            string url = CreateFacetOptionsUrl(section, facetGroupName);
+            Dictionary<string, string> requestHeaders = new Dictionary<string, string>();
+            AddAuthHeaders(this.Options, requestHeaders);
+
+            string result;
+            try
+            {
+                result = await MakeHttpRequest(this.Options, HttpMethod.Post, url, requestHeaders, facetOption);
+            }
+            catch (Exception e)
+            {
+                throw new ConstructorException(e);
+            }
+
+            if (result != null)
+            {
+                return JsonConvert.DeserializeObject<FacetOption>(result);
+            }
+
+            throw new ConstructorException("CreateFacetOption response data is malformed");
+        }
+
+        /// <summary>
+        /// Gets all Facet Options in a facet group.
+        /// <param name="facetGroupName">Facet Group to retrieve the Facet Options from.</param>
+        /// <param name="pageRequest">PageRequest object for pagination.</param>
+        /// <param name="section">Section in which the facet is defined.</param>
+        /// </summary>
+        public async Task<FacetOptionsGetAllResponse> GetAllFacetOptions(string facetGroupName, [Optional] PageRequest pageRequest, [Optional] string section)
```

Comment:
Ooh `PaginationOptions` is definitely a better name. I figured it might be re-usable too so I gave it it's own class in case there's other methods / checks (inherited or otherwise) that we might want to associate with them. Open to both ways though.

Reply:
I'm happy with either direction. cc @crgee1 in case you have any opinions on this